sktime/forecasting/tests/test_all_forecasters.py

```diff
@@ -230,6 +230,12 @@ def test_predict_time_index(
             _assert_correct_pred_time_index(y_pred.index, cutoff, fh=fh_int)
             _assert_correct_columns(y_pred, y_train)
 
+            if estimator_instance.get_tag("capability:pred_int"):
```

Comment:
clarification: if this check is specifically made, what can cause `NotImplementedError`?

Reply:
I thought something about in-sample or out-of-sample.

But you are right the try/except is a block with an unclear condition, thinking carefully, I do not know for sure why it is there!

So, let's see what happens when we remove it:
https://github.com/sktime/sktime/pull/4476
If it has a good reason to break in cases on `main`, we should probably replace it with a new tag.

Reply:
I assume we consider this addressed by #4476